kafka/src/main/java/com/metamx/tranquility/kafka/writer/WriterController.java

```diff
@@ -0,0 +1,107 @@
+/*
+ * Licensed to Metamarkets Group Inc. (Metamarkets) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  Metamarkets licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package com.metamx.tranquility.kafka.writer;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.base.Throwables;
+import com.metamx.common.logger.Logger;
+import com.metamx.tranquility.kafka.model.TopicSchemaPair;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.regex.Pattern;
+
+/**
+ * Manages the creation and operation of EventWriters based on a Kafka topic to Druid batch ingestion spec mapping.
+ */
+public class WriterController
+{
+  private static final Logger log = new Logger(WriterController.class);
+
+  private String zkConnect;
+  private String indexService;
+  private String discoveryPath;
+  private List<TopicSchemaPair> topicSchemaPairs;
+  private Map<String, SynchronizedEventWriter> writers = new HashMap<>();
+
+  public WriterController(String zkConnect, String indexService, String discoveryPath, String schemaMap) {
+    this.zkConnect = zkConnect;
+    this.indexService = indexService;
+    this.discoveryPath = discoveryPath;
+
+    try {
```

Comment:
It'd be nicer to pass in the actual list of TopicSchemaPairs rather than a string that gets deserialized. Generally we try to give classes like this what they actually need, and handle the instantiation of it somewhere else (either the Main class or a factory or something like that)